pkg/loop/generic_service.go

```diff
@@ -0,0 +1,49 @@
+package loop
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+	"google.golang.org/grpc"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/logger"
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+	"github.com/smartcontractkit/chainlink-relay/pkg/utils"
+)
+
+var _ ocrtypes.ReportingPluginFactory = (*GenericService)(nil)
+
+// GenericService is a [types.Service] that maintains an internal [types.PluginGeneric].
+type GenericService struct {
```

Comment:
The modifier `Generic` doesn't feel right to me. Maybe just because the overloading with generic types? Is there another term we could use? `Dynamic`? :thinking:  Or even just drop it in some cases to say only `Plugin`?

Reply:
Yeah finding a name for this was really tricky. I'll try PluginService.